app/grandchallenge/cases/admin.py

```diff
@@ -102,7 +111,72 @@ def algorithm(self, obj):
             return obj.algorithm_result.job.algorithm_image.algorithm
 
 
+class DownloadableFilter(admin.SimpleListFilter):
+    """Allow filtering on downloadable files."""
+
+    title = "Downloadable"
+    parameter_name = "downloadable"
+
+    def lookups(self, request, model_admin):
+        return (("yes", "Yes"),)
+
+    def queryset(self, request, queryset):
+        if self.value() == "yes":
+            return queryset.filter(staged_file_id__isnull=False)
+        return queryset
+
+
+class RawImageFileAdmin(admin.ModelAdmin):
+    list_filter = (DownloadableFilter,)
+    list_display = ("filename", "upload_session", "download")
+    readonly_fields = ("download",)
+
+    def download(self, instance):
+        if not instance.staged_file_id:
+            return
+        return format_html(
+            f'<a class="button" href={reverse(f"admin:{self.model._meta.app_label}_{self.model._meta.model_name}_download", kwargs={"object_id": instance.pk})}>Download</a>'
+        )
+
+    def download_view(self, request, object_id, **kwargs):
+        obj = self.get_object(request, unquote(object_id), None)
+        if not self.has_view_or_change_permission(request, obj):
+            raise PermissionDenied
+
+        try:
+            saf = StagedAjaxFile(obj.staged_file_id).open()
+            response = HttpResponse(
+                saf.read(), content_type="application/dicom"
+            )
+            response[
+                "Content-Disposition"
+            ] = f'attachment; filename="{obj.filename}"'
+            return response
+        except Exception:
+            raise Http404("File not found")
+
+    def get_urls(self):
+        def wrap(view):
+            def wrapper(*args, **kwargs):
+                return self.admin_site.admin_view(view)(*args, **kwargs)
+
+            wrapper.model_admin = self
+            return update_wrapper(wrapper, view)
+
+        urls = super().get_urls()
+
+        # Replace catchall with download endpoint, as the catchall is only there
+        # for django < 1.10 compatibility.
```

Comment:
I don't understand this comment..?

Reply:
Agreed, it is a bit unclear. Come to think of it, this probably isn't the best way to go about this anyway, as it relies on the code in the super function to not change.